core/devmode/src/main/java/io/quarkus/dev/DevModeMain.java

```diff
@@ -110,6 +114,36 @@ public void start() throws Exception {
         }
     }
 
+    // copies the .env file to the directory where the process is running
+    // this is done because the for the .env file to take effect, it needs to be
+    // in the same directory as the running process
+    private void copyDotEnvFile() {
+        File projectDir = context.getProjectDir();
+        if (projectDir == null) { // this is the case for QuarkusDevModeTest
+            return;
+        }
+        Path currentDir = Paths.get("").toAbsolutePath().normalize();
+        Path dotEnvPath = projectDir.toPath().resolve(".env");
+        if (Files.exists(dotEnvPath)) {
+            try {
+                Path link = currentDir.resolve(".env");
+                silentDeleteFile(link);
+                // create a symlink to ensure that user updates to the file have the expected effect in dev-mode
+                Files.createSymbolicLink(link, dotEnvPath);
```

Comment:
Does it work on Windows?

Reply:
I suppose so as you have a test.

Reply:
Yup, thankfully we have CI running on Windows as well 😎